compiler/rustc_ast_lowering/src/item.rs

```diff
@@ -384,17 +384,18 @@ impl<'hir> LoweringContext<'_, 'hir> {
                     AnonymousLifetimeMode::CreateParameter,
                     |this, _| {
                         let trait_ref = trait_ref.as_ref().map(|trait_ref| {
-                            this.lower_trait_ref(trait_ref, ImplTraitContext::disallowed())
-                        });
+                            let trait_ref =
+                                this.lower_trait_ref(trait_ref, ImplTraitContext::disallowed());
 
-                        if let Some(ref trait_ref) = trait_ref {
                             if let Res::Def(DefKind::Trait, def_id) = trait_ref.path.res {
                                 this.trait_impls
                                     .entry(def_id)
                                     .or_default()
                                     .push(lowered_trait_impl_id);
                             }
-                        }
+
+                            trait_ref
+                        });
```

Comment:
Actually, what is the point of this? There's no reason to change control flow here is there?

Reply:
I actually don't remember. I *think* I was trying something out and I forgot about it. Thanks!

Reply:
Also, this should probably just be `if let Some(def_id) = trait_ref.def_id()`. Otherwise it skips trait impls. Not sure who to ping about this though, and it shouldn't be changed in this PR.